### ISSUE TITLE
invalidate or delete half-done files if a processing step fails

### ISSUE THREAD

**Issue by emiltin** [Project-OSRM/osrm-backend#1756]
if a processings step (eg osrm-extract) runs out of memory or otherwise fails, half-finished files can will be left. further processing steps, eg. osrm-prepare, will usually choke on these half-baked files.

it would be useful to mark files once a processing step completes succesfully, so futher steps don't try to use invalid data, or simply delete all half-finished files in case a processing step fails.

**Comment by TheMarex** [comment]
Yeah good idea. We could simply write out a canary at the end of the file and check that.

**Comment by daniel-j-h** [comment]
Why not make it simple and as it is usually done for "transactional" I/O:
write to a (possibly "hidden") temporary file, on success rename.
